gnomad_qc/v3/create_release/sanity_checks.py

```diff
@@ -0,0 +1,711 @@
+
+
+def vcf_field_check(
+    t: Union[hl.MatrixTable, hl.Table],
+    header_dict: Dict[str, Dict[str, Dict[str, str]]],
+    row_annotations: List[str],
+    hists: List[str] = HISTS,
+) -> bool:
+    """
+    Check that all VCF fields and descriptions are present in input Table and VCF header dictionary.
+
+    :param hl.MatrixTable, hl.Table t: Input MatrixTable or Tableto be exported to VCF.
+    :param Dict[str, Dict[str, Dict[str, str]]] header_dict: VCF header dictionary.
+    :param List[str] row_annotations: List of row annotations in MatrixTable.
+    :param List[str] hists: List of variant histogram annotations. Default is HISTS.
+    :return: Bool with whether all expected fields and descriptions are present.
+    :rtype: bool
+    """
+    t = t.rows() if isinstance(t, hl.MatrixTable) else t
+
+    # Confirm all VCF fields/descriptions are present before exporting
+    hist_fields = []
+    for hist in hists:
+        hist_fields.extend(
+            [
+                f"{hist}_bin_freq",
+                f"{hist}_n_smaller",
+                f"{hist}_n_larger",
+                f"{hist}_raw_bin_freq",
+                f"{hist}_raw_n_smaller",
+                f"{hist}_raw_n_larger",
+            ]
+        )
+
+    missing_fields = []
+    missing_descriptions = []
+    for item in ["info", "filter"]:
+        if item == "info":
+            annots = row_annotations
+        else:
+            annot_t = t.explode(t.filters)
+            annots = list(annot_t.aggregate(hl.agg.collect_as_set(annot_t.filters)))
+
+        temp_missing_fields = []
+        temp_missing_descriptions = []
+        for field in annots:
+            try:
+                description = header_dict[item][field]
+                if len(description) == 0:
+                    logger.warning(
+                        f"{field} in T info field has empty description in VCF header!"
+                    )
+                    temp_missing_descriptions.append(field)
+            except KeyError:
+                logger.warning(f"{field} in T info field does not exist in VCF header!")
+                # NOTE: some hists are not exported, so ignoring here
+                # END entry is also not exported (removed during densify)
+                if (field not in hist_fields) and (field != "END"):
+                    temp_missing_fields.append(field)
+
+        missing_fields.extend(temp_missing_fields)
+        missing_descriptions.extend(temp_missing_descriptions)
+
+    if len(missing_fields) != 0 or len(missing_descriptions) != 0:
+        logger.error(
+            "Some fields are either missing or missing descriptions in the VCF header! Please reconcile."
+        )
+        logger.error(f"Missing fields: {missing_fields}")
+        logger.error(f"Missing descriptions: {missing_descriptions}")
+        return False
+
+    logger.info("Passed VCF fields check!")
+    return True
+
+
+def sanity_check_release_t(
```

Comment:
Same comment about adding to gnomad_methods. Possibly with options for each of the sanity checks?